pkg/scheduler/framework/plugins/noderesources/requested_to_capacity_ratio.go

```diff
@@ -123,54 +123,8 @@ func (pl *RequestedToCapacityRatio) ScoreExtensions() framework.ScoreExtensions
 	return nil
 }
 
-func validateFunctionShape(shape functionShape) error {
-	if len(shape) == 0 {
-		return fmt.Errorf("at least one point must be specified")
-	}
-
-	for i := 1; i < len(shape); i++ {
-		if shape[i-1].utilization >= shape[i].utilization {
-			return fmt.Errorf("utilization values must be sorted. Utilization[%d]==%d >= Utilization[%d]==%d", i-1, shape[i-1].utilization, i, shape[i].utilization)
-		}
-	}
-
-	for i, point := range shape {
-		if point.utilization < minUtilization {
-			return fmt.Errorf("utilization values must not be less than %d. Utilization[%d]==%d", minUtilization, i, point.utilization)
-		}
-		if point.utilization > maxUtilization {
-			return fmt.Errorf("utilization values must not be greater than %d. Utilization[%d]==%d", maxUtilization, i, point.utilization)
-		}
-		if point.score < minScore {
-			return fmt.Errorf("score values must not be less than %d. Score[%d]==%d", minScore, i, point.score)
-		}
-		if int64(point.score) > maxScore {
-			return fmt.Errorf("score values not be greater than %d. Score[%d]==%d", maxScore, i, point.score)
-		}
-	}
-
-	return nil
-}
-
-func validateResourceWeightMap(resourceToWeightMap resourceToWeightMap) error {
-	if len(resourceToWeightMap) == 0 {
-		return fmt.Errorf("resourceToWeightMap cannot be nil")
-	}
-
-	for resource, weight := range resourceToWeightMap {
-		if weight < 1 {
-			return fmt.Errorf("resource %s weight %d must not be less than 1", string(resource), weight)
-		}
-	}
-	return nil
-}
-
 func buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape functionShape, resourceToWeightMap resourceToWeightMap) func(resourceToValueMap, resourceToValueMap, bool, int, int) int64 {
 	rawScoringFunction := buildBrokenLinearFunction(scoringFunctionShape)
-	err := validateResourceWeightMap(resourceToWeightMap)
-	if err != nil {
-		klog.Error(err)
```

Comment:
Please not that this is now returned as error instead of just being logged as one.

Reply:
sg